### PR TITLE
made dcm to quat function always return principal quat

### PR DESCRIPTION
**Pull request by ryanelandt**
The dcm to `Quat` function should always return the principal `Quat` otherwise commands like `SPQuat(RotMatrix(SPQuat(-0.01916, -0.28, -0.42)))` will produce ill-conditioned `SPQuat`. The added time is around 0.4 ns on my computer.

Messing around with the spacing near the bottom of the `rotation_tests.jl` file was done by ATOM.

### CODE REVIEW

**Comment by tkoolen** [comment]
I don't fully understand the problem. What's ill-conditioned about this case?

```julia
julia> using Rotations

julia> spq = SPQuat(-0.01916, -0.28, -0.42)
3×3 SPQuat{Float64}(-0.01916, -0.28, -0.42):
 -0.293857  0.821506  -0.488646
 -0.767022  0.102389   0.633398
  0.570373  0.560931   0.600026

julia> spq2 = SPQuat(RotMatrix(spq))
3×3 SPQuat{Float64}(0.075088, 1.09732, 1.64598):
 -0.293857  0.821506  -0.488646
 -0.767022  0.102389   0.633398
  0.570373  0.560931   0.600026

julia> spq3 = principal_value(spq2)
3×3 SPQuat{Float64}(-0.01916, -0.28, -0.42):
 -0.293857  0.821506  -0.488646
 -0.767022  0.102389   0.633398
  0.570373  0.560931   0.600026

julia> [spq.x, spq.y, spq.z] - [spq3.x, spq3.y, spq3.z]
3-element Array{Float64,1}:
 -6.938893903907228e-18
  0.0
 -1.1102230246251565e-16
```

That all seems very reasonable to me.

Also this would make the dcm -> quat mapping non-differentiable, which I'm not sure is a good idea; it seems like it would result in issues similar to #44.

**Comment by ryanelandt** [comment]
It gives the more ill-conditioned `SPQuat` and isn't what one would expect when getting an `SPQuat` from a rotation matrix. The dcm -> quat mapping isn't differentiable as is (see code below). I don't think it can be mode to be differentiable without giving a "reference" `Quat` and having the function the one closest to it.

```julia
julia> quat_0 = Quat(RotMatrix( eps(), -1.0, 0.0, 1.0, -eps(), 0.0, 0.0, 0.0, 1.0))
3×3 Quat{Float64}(0.707107, 0.0, 0.0, -0.707107):
  0.0  1.0   0.0
 -1.0  0.0  -0.0
 -0.0  0.0   1.0

julia> quat_1 = Quat(RotMatrix(-eps(), -1.0, 0.0, 1.0, -eps(), 0.0, 0.0, 0.0, 1.0))
3×3 Quat{Float64}(-0.707107, 0.0, 0.0, 0.707107):
 -3.33067e-16   1.0          0.0
 -1.0          -3.33067e-16  0.0
  0.0           0.0          1.0
```

**Comment by tkoolen** [comment]
> more ill-conditioned `SPQuat`

To be completely clear, do you consider `spq2` in my example above to be an ill-conditioned result just because it's not a principal value? I was expecting a catastrophic numerical issue, but this can't get arbitrarily bad, can it?

> The dcm -> quat mapping isn't differentiable as is

OK, thanks for the example, point taken.

**Comment by ryanelandt** [comment]
Yes, I should have said that it is not a principal value. I think you're right about it not being able to get arbitrarily bad; I don't see a way it can.

**Comment by tkoolen** [comment]
I guess I'm OK with the change. Neither the pros nor the cons seem very compelling. Let's leave this open for others to comment on for a bit.

**Comment by andyferris** [comment]
I'm usually ambivalent to these kinds of changes - if it makes a practical improvement to someone's workflow then that seems sufficiently compelling.

**Comment by hyrodium** [comment]
Is this PR just waiting resolving the conflicts?

**Comment by ryanelandt** [comment]
> Is this PR just waiting resolving the conflicts?

It's open to hear people's thoughts. There may not be many people who use `MRP`s. If you don't have any, I can close this issue.

**Comment by hyrodium** [comment]
This gif shows parameters of `MRP`, which is generated by `ps = Rotations.params.(MRP.(RotMatrix{3}.(rand(UnitQuaternion, N))))`.

MRP

This is a same shape as 4 cells of [stereographically procjected hypercube](https://youtu.be/Eef3twD0N94?t=285).

These cells were generated coincidentally because of [the if statement](https://github.com/JuliaGeometry/Rotations.jl/blob/v1.0.4/src/unitquaternion.jl#L81), and I feel this is really interesting.
However this is not user friendly, and I think this should be fixed for principal value. And also, there's not much computational cost for getting principal value.

If we return principal value, then the visualized points will be inside a unit-ball.

**Comment by ryanelandt** [comment]
I agree with your thoughts @hyrodium.

Without principal value, this part of Rotations is numerically interesting, but not user friendly.

**Comment by hyrodium** [comment]
@ryanelandt
Could you resolve the conflicts? After that, I can merge this PR.

File: test/rotation_tests.jl [hunk 1]
```diff
@@ -259,11 +259,12 @@ all_types = (RotMatrix{3}, AngleAxis, RotationVec,
 
     @testset "DCM to UnitQuaternion" begin
         pert = 1e-3
-		quat = UnitQuaternion
```
Comment (hyrodium) [inline test/rotation_tests.jl:262]: I found this indent strange, but that was just `<tab>`.

**Pull request merged**